api/email/CloudEmail.js

```diff
@@ -6,46 +6,47 @@
 */
 
 
-module.exports = function() {   
-
-   /**
-    *Description : Send Email to all users in the selected aplication
-    *Params: 
-    *- Param secureKey: Secure key of System
-    *Returns:
-    -Success : success on emails sent
-    -Error : Error Data( 'Server Error' : status 500 )
-    */
+module.exports = function () {
+
+    /**
+     *Description : Send Email to all users in the selected aplication
+     *Params: 
+     *- Param secureKey: Secure key of System
+     *Returns:
+     -Success : success on emails sent
+     -Error : Error Data( 'Server Error' : status 500 )
+     */
     global.app.post('/email/:appId/campaign', function (req, res) {
         console.log("++++ Email Campaign ++++++");
-        try{
+        try {
             var appId = req.params.appId;
             var appKey = req.body.key;
+            var query = req.body.query;
             var emailBody = req.body.emailBody;
             var emailSubject = req.body.emailSubject;
 
-            global.appService.isMasterKey(appId,appKey).then(function (isMasterKey) {
+            global.appService.isMasterKey(appId, appKey).then(function (isMasterKey) {
                 if (isMasterKey) {
-                    global.emailService.sendEmail(appId,emailBody,emailSubject,isMasterKey).then(function(data){
+                    global.emailService.sendEmail(appId, query, emailBody, emailSubject, isMasterKey).then(function (data) {
```

Comment:
query param should be before isMasterKey (as this param is not used most of the time) .